shacl/register.shacl.ttl

```diff
@@ -0,0 +1,198 @@
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix schema: <http://schema.org/> .
+@prefix dcat: <http://www.w3.org/ns/dcat#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+@prefix dc: <http://purl.org/dc/terms/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix reg: <http://terms.netwerkdigitaalerfgoed.nl/ns/register#> .
+
+##
+## Schema.org
+##
+
+#
+# Class shapes
+#
+
+reg:DatacatalogShape
+	a sh:NodeShape ;
+	sh:name "Datacatalogus"@nl, "Datacatalog"@en ;
+	rdfs:comment "Een datacatalogus bestaat uit een set van datasetbeschrijvingen"@nl ;
+	sh:targetClass schema:DataCatalog ;
+	sh:property reg:DatatalogProperty ;
+.
+
+reg:DatasetShape
+	a sh:NodeShape ;
+	sh:name "Datasetbeschrijving"@nl, "Datasetdescription"@en ;
+	rdfs:comment "Een datasetbeschrijving bestaat uit een elementen die de dataset beschrijft"@nl ;
+	sh:and (
+		[ sh:targetClass schema:Dataset ; ]
+		[ sh:not ( [ sh:targetClass schema:WebPage ; ]  ) ]	
```

Comment:
:+1:

Reply:
No, it doesn't work. But the thing with target selection it just doesn't target a node and skips all associated validation. Hard to notice (use shacl validate --debug to see targeted nodes).

This construct is for the KB NBT case in which a mainEntityOfPage contains a node with is a Dataset and a WebPage. I'm looking for a construct that targets Datasets but not if it's also a WebPage. The following custom target doesn't work either (no Datasets are targeted at all):
```
  sh:target [
    a sh:SPARQLTarget ;
    sh:select """ SELECT ?this WHERE { ?this a <http://schema.org/Dataset> ; MINUS { ?node a <http://schema.org/WebPage> } } """ ;
  ] ;

```